velox/dwio/common/Reader.cpp

```diff
@@ -211,7 +176,7 @@ void RowReader::readWithRowNumber(
     flatRowNum = rowNumVector->asUnchecked<FlatVector<int64_t>>();
   }
   auto* rawRowNum = flatRowNum->mutableRawValues();
-  if ((numChildren == numConstChildren) && !hasDeletion(mutation)) {
+  if ((numChildren == numNotReadFromFileChildren) && !hasDeletion(mutation)) {
```

Comment:
You don't need to pass all these 3 to the function.  Just do
```c++
const bool contiguousRowNumbers = (numChildren == numNotReadFromFileChildren) && !hasDeletion(mutation);
// ...
fillRowNumberVector(rowNumVector, contiguousRowNumbers, previousRow, rowsToRead, columnReader, size, pool);
```

Reply:
I found `mutation` may be changed in `columnReader->next`, so I add `const bool contiguousRowNumbers = (numChildren == numNotReadFromFileChildren);`  and use it in function `fillRowNumberVector` as `contiguousRowNumbers && !hasDeletion(mutation)`.